### ISSUE TITLE
Add support for conditional type nodes

### ISSUE THREAD

**Issue by kayahr** [vega/ts-json-schema-generator#100]
Because of https://github.com/YousefED/typescript-json-schema/issues/173 I'm currently trying to migrate to `ts-json-schema-generator`. The first problem I have with the new tool is missing support for conditional type nodes which worked fine with the `typescript-json-schema` tool. Here is a minimal example:

```typescript
export type TypeName<T> =
    T extends string ? "string" :
    T extends number ? "number" :
    T extends boolean ? "boolean" :
    "object";

export type TestJSON = TypeName<string>;
```

When I try to generate the schema for `TestJSON` with this tool then I get this error message:

```
UnknownNodeError: Unknown node "TypeName<string>" (ts.SyntaxKind = 176)
```

I really like the clean structure of the new tool and it looks pretty straight forward how to implement this feature by adding a `ConditionalTypeNodeParser` class like this:

```typescript
export class ConditionalTypeNodeParser implements SubNodeParser {
    public constructor(
        private typeChecker: ts.TypeChecker,
        private childNodeParser: NodeParser,
    ) {
    }

    public supportsNode(node: ts. ConditionalTypeNode): boolean {
        return node.kind === ts.SyntaxKind.ConditionalType;
    }

    public createType(node: ts.ConditionalTypeNode, context: Context): BaseType {
        ... magic here ...
    }
}
```

But unfortunately I lack the skills to implement the `createType` method. In theory the method has to check if `node.checkType` is assignable to `node.extendsType` and then returns `node.trueType` or `node.falseType`. But I don't know if this assignability check must be written from scratch here or if TypeScript already exposes some functionality to do this. And it gets more complicated if the check type is a union type so `TypeName<string | Date>` is resolved to `"string" | "object"` in this example.

**Comment by domoritz** [comment]
This is closely related to https://github.com/vega/ts-json-schema-generator/issues/71

**Comment by kayahr** [comment]
I gave it a try and implemented somewhat working conditional type support. It works for the simple stuff I need and a few more complex use cases and also for basic exclude types but there are also type combinations which don't work at all... I'm not really happy with this implementation so I'm not sure if a pull request makes sense. It feels to complicated and I'm still not sure if it isn't possible to use some TypeScript internals to do all this... But maybe it is an acceptable "good enough for now" solution?

https://github.com/vega/ts-json-schema-generator/compare/master...iplabs:feature/conditional-type?expand=1

I also added some tests which shows what is already working.
